lms/djangoapps/discussion_api/api.py

```diff
@@ -361,21 +366,52 @@ def get_comment_list(request, thread_id, endorsed, page, page_size):
     return get_paginated_data(request, results, page, num_pages)
 
 
-def _check_editable_fields(cc_content, data, context):
+def _check_fields(allowed_fields, data, message):
     """
-    Raise ValidationError if the given update data contains a field that is not
-    in editable_fields.
+    Raise ValidationError if the given data contains a key that is not in
+    allowed_fields
     """
-    editable_fields = get_editable_fields(cc_content, context)
-    non_editable_errors = {
-        field: ["This field is not editable."]
-        for field in data.keys()
-        if field not in editable_fields
-    }
+    non_editable_errors = {field: [message] for field in data.keys() if field not in allowed_fields}
     if non_editable_errors:
         raise ValidationError(non_editable_errors)
 
 
+def _check_initializable_thread_fields(data, context):  # pylint: disable=invalid-name
+    """
+    Raise ValidationError if the given data contains a thread field that is not
+    initializable by the requesting user
+    """
+    _check_fields(
+        get_initializable_thread_fields(context),
+        data,
+        "This field is not initializable."
+    )
+
+
+def _check_initializable_comment_fields(data, context):  # pylint: disable=invalid-name
+    """
+    Raise ValidationError if the given data contains a thread field that is not
+    initializable by the requesting user
+    """
+    _check_fields(
+        get_initializable_comment_fields(context),
+        data,
+        "This field is not initializable."
+    )
+
+
+def _check_editable_fields(cc_content, data, context):
+    """
+    Raise ValidationError if the given update data contains a field that is not
+    editable by the requesting user
+    """
+    _check_fields(
+        get_editable_fields(cc_content, context),
+        data,
+        "This field is not editable."
+    )
+
+
```

Comment:
These functions (`_check_{...}_fields`) are redundant, but you mentioned that you were probably going to refactor them at some point in the future, so that's fine